app/api/chat/route.ts

```diff
@@ -16,24 +69,103 @@ export async function POST(request: Request) {
   const title =
     message.length > 40 ? message.slice(0, 40).concat("...") : message;
 
-  const saveChatResult = await saveChat({
+  const [chat] = await saveChat({
     userId: session.user.id,
     title,
+    marketAddress: marketId,
   });
-  const chat = saveChatResult[0];
 
   await saveMessage({
     chatId: chat.id,
     message: { response: message },
     role: "user",
   });
 
+  try {
+    if (marketId) createMarketChat(message, chat, marketId);
+    else verifyAndCreateChat(message, chat);
+  } catch (error) {
+    let message;
+    if (error instanceof Error) message = error.message;
+    else message = String(error);
+
+    return new Response(message, { status: 400 });
+  }
+
   return NextResponse.json({ chatId: chat.id });
 }
 
+async function createMarketChat(
```

Comment:
I think create `createMarketChat` should create a market chat and not create a chat if no market.

The condition should be outside of this function.

Reply:
I will update names. Let me know if they are better.